jbpm-task-assigning/jbpm-task-assigning-core/src/main/java/org/jbpm/task/assigning/model/Task.java

```diff
@@ -0,0 +1,333 @@
+import org.optaplanner.core.api.domain.variable.PlanningVariableReference;
+
+/**
+ * Task is the only planning entity that will be changed during the problem solving, and we have only one
+ * PlanningVariable.
+ * <p>
+ * In this particular problem we want to do the assignments in a way that the list of tasks
+ * for a given user is ordered. (see e.g. that Higher priority tasks must be resolved first is a soft constraint)
+ * <p>
+ * User1: A <- B <- C <- D
+ * <p>
+ * User2: E <- F
+ * <p>
+ * The initial task of each sequence really points to the user that will own all the tasks in the list, so when a
+ * solution is created we'll really have something like this.
+ * <p>
+ * User1 <- A <- B <- C <- D  (In this example we say that User1 the anchor)
+ * <p>
+ * This explains why property "previousTaskOrUser" can be assigned with User or a Task.
+ * <p>
+ * BUT the solver will build the solutions in a way that only the first item of the "chain" will point to
+ * a user. This is how a CHAINED configuration works. And the way the solver knows which must be the fact class
+ * that must be used for setting the previousTaskOrUser property is by considering the order in the PlanningVariable
+ * configuration.
+ * @PlanningVariable(valueRangeProviderRefs = {"userRange", "taskRange"}, graphType = PlanningVariableGraphType.CHAINED)
+ * <p>
+ * Here we basically declared that we want to build a CHAINED graph but also since the "userRange" is the first
+ * valueRangeProviderRef we are declaring that the User will be used as anchor.
+ * So the solver will always start by using an User as the head of the list and then it'll consider adding Tasks
+ * to the linked structure.
+ * <p>
+ * Additionally for the calculation of the scores, etc, given a Task of a particular solution we'll want to know quickly
+ * which is the User that was assigned to this task.
+ * <p>
+ * If we have the following chain
+ * <p>
+ * Employee1 (the anchor) <- A <- B <- C <- D
+ * <p>
+ * and we take e.g. D, then to get the assigned Employee1 the list must be iterated, but this probably not the best idea.
+ * <p>
+ * Solution, add a shadow variable and let the solver populate this variable when the solution is constructed.
+ * With the declaration below a shadow variable is defined for keeping a reference to the anchor. This shadow variable
+ * is populated and kept consistent by the solver.
+ * <p>
+ * Shadow variable:
+ * Let all Tasks have a reference to the anchor of the chain, the assigned user.
+ * @AnchorShadowVariable(sourceVariableName = "previousTaskOrUser")
+ * private User user;
+ * <p>
+ * CustomShadowVariable startTime:
+ * Additionally is a convenient shadow variable is declared for having the startTime of a task already calculated.
+ * @CustomShadowVariable(variableListenerClass = StartTimeUpdatingVariableListener.class,
+ * // Arguable, to adhere to API specs (although this works), nextTask and user should also be a source,
```

Comment:
Haven't seen this comment when reviewing. 
Well, this comment present in the other places in optaplanner when this chained graph structure is being used. So thought it could make sense to keep it. If not it can be removed.